File: 백준/Bronze/10798. 세로읽기/세로읽기.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
const max = Math.max(...input.map(e => e.length));
const words = [];

for (let i = 0; i < max; i++) {
    const result = [];
    
    for (let j = 0; j < 5; j++) {
        result.push(input[j][i]);
    }
    
    words.push(result.join(''));
}

console.log(words.join(''));

File: 백준/Bronze/10798. 세로읽기/세로읽기.py
import sys

board = list(map(lambda row: row.rstrip(), sys.stdin.readlines()))
w = max(map(lambda row: len(row), board))
h = len(board)
result = ''

for c in range(0, w):
    for r in range(0, h):
        if (len(board[r]) > c):
            result += board[r][c]

print(result)